Properly handle game pause in VideoPlayer

KoBeWi opened the pull request:
Fixes #33702

Xrayez commented on scene/gui/video_player.cpp:
```diff
@@ -186,6 +186,18 @@ void VideoPlayer::_notification(int p_notification) {
+		} break;
+
+		case NOTIFICATION_UNPAUSED: {
+			playback->set_paused(false);
+			set_process_internal(true);
```
Does it mean that the playback is going to be resumed even if a video was paused manually (`$video.paused = true`) by a user? (not via `get_tree().paused`).

YeldhamDev replied:
Xrayez is right, it should differentiate if it was paused manually or not before unpausing it.

KoBeWi committed:
Properly handle game pause in VideoPlayer

KoBeWi commented:
Rebased and reworked. I added `paused_from_tree` variable, which is set to true when the player is playing and gets paused from scene tree. I couldn't test if it works though, because VideoStreamPlayer doesn't seem functional, idk

akien-mga commented on scene/gui/video_stream_player.cpp:
```diff
@@ -177,8 +177,30 @@ void VideoStreamPlayer::_notification(int p_notification) {
+				if (playback.is_valid()) {
+					playback->set_paused(true);
+					set_process_internal(false);
+				};
```
Trailing semicolon (also in `UNPAUSED`).

KoBeWi commented:
Tested and it works correctly.

akien-mga commented:
Thanks!
